asserts/asserts.go

```diff
@@ -331,15 +331,17 @@ var _ Assertion = (*assertionBase)(nil)
 // where:
 //
 //    HEADER is a set of header entries separated by "\n"
-//    BODY can be arbitrary,
+//    BODY can be arbitrary text,
 //    SIGNATURE is the signature
 //
+// Both BODY and HEADER must be UTF8.
+//
 // A header entry for a single line value (no '\n' in it) looks like:
 //
 //   NAME ": " SIMPLEVALUE
 //
 // The format supports multiline text values (with '\n's in them) and
-// lists possibly nested with string scalars in them.
+// lists or maps possibly nested with string scalars in them.
```

Comment:
Maybe, "The format supports multiline text values (with '\n's in them) and lists or maps, possibly nested, with string scalars in them". I think the extra commas make the "possibly nested" apply to "list or maps" a little better and also make it clearer what contains scalars.